package ab55;


public class Exceptionhandling {

	public static void main(String[] args) {
		 
		int a=1/1;
		int b=0/1;
		
		try {
		int c=1/1;//1 coming from parent class ,0 from grand parent
		}
		catch(ArithmeticException a1)
		{
		
			System.out.println("Handled the exception");
		}
		
		//System.out.println("hello");
		finally
		{
			System.out.println("I will always execute");
			
			
		}
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		/*Scanner s1=new Scanner(System.in);
		
		try {
		int array1[]=new int[s1.nextInt()];
		
		
		}
		catch(InputMismatchException c1)
		{
			
			System.out.println("Handled the exception 1");
		}
		catch(NegativeArraySizeException c2)
		{
			
			System.out.println("Handled the exception 2");
		}
		
		
		
		*/
		
		
		
		
		
		
		
		
		
		
		
		
		

	}
		
		
		
		
		
		
		
		
		/*int a=1/1;
		int b=0/1;
		try 
		{
			
		int c=1/0;
		
		}

		catch(ArithmeticException)
		
		*/
		
		
		
		
		
		
		
		
				
		
	}


